internal/storage/v2/clickhouse/tracestore/dbmodel/from.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package dbmodel
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/internal/jptrace"
+	"github.com/jaegertracing/jaeger/internal/telemetry/otelsemconv"
+)
+
+// FromRow converts a ClickHouse stored span row to an OpenTelemetry Traces object.
+func FromRow(storedSpan *SpanRow) ptrace.Traces {
+	trace := ptrace.NewTraces()
+	resourceSpans := trace.ResourceSpans().AppendEmpty()
+	scopeSpans := resourceSpans.ScopeSpans().AppendEmpty()
+	span := scopeSpans.Spans().AppendEmpty()
+
+	sp, err := convertSpan(storedSpan)
+	sp.CopyTo(span)
+	if err != nil {
+		jptrace.AddWarnings(span, err.Error())
+	}
+
+	resource := resourceSpans.Resource()
+	rs := convertResource(storedSpan)
+	rs.CopyTo(resource)
+
+	scope := scopeSpans.Scope()
+	sc := convertScope(storedSpan)
+	sc.CopyTo(scope)
+
+	return trace
+}
+
+func convertResource(sr *SpanRow) pcommon.Resource {
+	resource := ptrace.NewResourceSpans().Resource()
+	resource.Attributes().PutStr(otelsemconv.ServiceNameKey, sr.ServiceName)
+	// TODO: populate attributes
```

Comment:
would this be different from calling putAttributes? Same question for scope below.